stl/inc/flat_map

```diff
@@ -1428,74 +1490,82 @@ private:
         {
             _Key _Key_to_insert(_STD forward<_KeyTy>(_Key_val));
             _Mapped _Mapped_to_insert(_STD forward<_MappedTy>(_Mapped_val));
-            this->_Insert_exact(this->cbegin() + _Index, _STD move(_Key_to_insert), _STD move(_Mapped_to_insert));
+            this->_Insert_exact(_STD cbegin(*this) + _Index, _STD move(_Key_to_insert), _STD move(_Mapped_to_insert));
         }
         return this->begin() + _Index;
     }
 };
 
-template <class _KeyContainer, class _MappedContainer, class _Compare = less<typename _KeyContainer::value_type>>
+template <_Not_allocator_for_container _KeyContainer, _Not_allocator_for_container _MappedContainer,
+    _Not_allocator_for_container _Compare = less<typename _KeyContainer::value_type>>
 flat_multimap(_KeyContainer, _MappedContainer, _Compare = _Compare())
     -> flat_multimap<typename _KeyContainer::value_type, typename _MappedContainer::value_type, _Compare, _KeyContainer,
         _MappedContainer>;
 
 template <class _KeyContainer, class _MappedContainer, class _Allocator>
 flat_multimap(_KeyContainer, _MappedContainer, _Allocator) -> flat_multimap<typename _KeyContainer::value_type,
     typename _MappedContainer::value_type, less<typename _KeyContainer::value_type>, _KeyContainer, _MappedContainer>;
-template <class _KeyContainer, class _MappedContainer, class _Compare, class _Allocator>
+template <_Not_allocator_for_container _KeyContainer, _Not_allocator_for_container _MappedContainer,
+    _Not_allocator_for_container _Compare, class _Allocator>

```

Comment:
The `flat_map` deduction guides use `_Valid_compare_for_container<_KeyContainer> _Compare, _Usable_allocator_for<_KeyContainer, _MappedContainer> _Allocator` while `flat_multimap` doesn't. They should be consistent.